proxy-node-shared/src/test/java/uk/gov/ida/notification/shared/ProxyNodeLoggerTest.java

```diff
@@ -1,26 +1,43 @@
 package uk.gov.ida.notification.shared;
 
+import ch.qos.logback.classic.Logger;
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import ch.qos.logback.core.Appender;
+import io.dropwizard.logging.LoggingUtil;
 import org.junit.Test;
 import org.junit.runner.RunWith;
-import org.mockito.Spy;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
 import org.mockito.junit.MockitoJUnitRunner;
+import org.slf4j.LoggerFactory;
 
-import java.util.logging.Level;
-
-import static org.mockito.AdditionalMatchers.not;
-import static org.mockito.ArgumentMatchers.contains;
-import static org.mockito.ArgumentMatchers.eq;
+import static org.assertj.core.api.Assertions.assertThat;
 import static org.mockito.Mockito.verify;
 
 @RunWith(MockitoJUnitRunner.class)
 public class ProxyNodeLoggerTest {
 
-    @Spy
+    @Mock
+    private Appender<ILoggingEvent> appender;
+
+    @Captor
+    private ArgumentCaptor<ILoggingEvent> loggingEventCaptor;
+
     private ProxyNodeLogger proxyNodeLogger = new ProxyNodeLogger();
 
     @Test
     public void shouldNotLogProxyNodeLoggerAsTheLogLocation() {
-        proxyNodeLogger.log(Level.INFO, "stop your messing around");
-        verify(proxyNodeLogger).addContext(eq(ProxyNodeMDCKey.LOG_LOCATION), not(contains(ProxyNodeLogger.class.getName())));
+        LoggingUtil.hijackJDKLogging();
+        Logger logger = (Logger) LoggerFactory.getLogger(ProxyNodeLogger.class);
+        logger.addAppender(appender);
+
+        proxyNodeLogger.info("test");
+
+        verify(appender).doAppend(loggingEventCaptor.capture());
+
+        final ILoggingEvent logEvent = loggingEventCaptor.getValue();
+        assertThat(logEvent.getMessage()).isEqualTo("test");
+        assertThat(logEvent.getMDCPropertyMap().get(ProxyNodeMDCKey.LOG_LOCATION.name())).doesNotContain(ProxyNodeLogger.class.getName());
```

Comment:
Should we also assert that `LOG_LOCATION.name()` is equal to what we expect? Otherwise the test still passes if this key is never actually set.

It's probably worth testing some of the other functionality in the PrroxyNodeLogger, like adding exception context,

Reply:
Added more tests 👍